src/Background.cpp

```diff
@@ -259,19 +253,30 @@ namespace Background {
 		std::unique_ptr<Color[]> colors(new Color[NUM_BG_STARS]);
 		std::unique_ptr<float[]> sizes(new float[NUM_BG_STARS]);
 		std::unique_ptr<float[]> brightness(new float[NUM_BG_STARS]);
+
+		// TODO: find a nicer solution for main menu
+
 		//fill the array
 		Uint32 num = 0;
 		if (space != nullptr && galaxy.Valid() && space->GetStarSystem() != nullptr) {
 			const SystemPath current = space->GetStarSystem()->GetPath();
 
-			const double size = 1.0;
-			const Sint32 visibleRadius = MathUtil::mix(BG_STAR_RADIUS_MIN, Clamp(m_visibleRadiusLy, BG_STAR_RADIUS_MIN, BG_STAR_RADIUS_MAX), Pi::GetAmountBackgroundStars()); // lyrs
+			constexpr float starsPerSector = 2.0; /* experimentally determined (already ajusted for that we are
```

Comment:
The potential sectors used are selected in a square, because that was easy, but are then rejected by distance leaving only stars within a spherical region.

Reply:
Also why pick only 2 stars per sectors when we have many more?

Reply:
The problem I encountered was that if `BG_STAR_RADIUS_MAX` was increased we'd run again in the issue that only half of the sky has stars. I also found that we can increase the visibleRadius far over the hard-coded 220 LY when selecting the maximum density, which I thought is nicer than filling up the rest with fake stars.

My idea was that we want to know, how big we need to make the `visibleRadius` to get approximately the number of stars we want (`NUM_BG_STARS`).

4/3 * pi * visibleRadius^3 = numSectors * sectorSize^3, i.e. I want the sum of the volume of all sectors be approximately the same as the volume of the sphere with radius `visibleRadius`. We can the numSectors by experimentally checking how many stars on average are in one sector. And I found that on average there are 2 stars in one sector (or rather 2 solar systems per sector). If we'd change the size of a sector this would break that's why I'm not sure if it's actually ok.

Reply:
Ah no, the fake stars are there because we never want to run the sector generator too much. It's incredibly expensive, so the populated stars are really only a shell close to the player, and anything further away is just ... padding to make it look more believable.

An extension to this idea is to eventually have the stars within jump distance be displayed on the night sky somehow, so again, that's only going to be stars close enough that the ship can jump too them.

Reply:
Oh ok, didn't think about that. What about if I cap the visible radius after calculating to BG_STAR_RADIUS_MAX or a parameter that can be changed? To be honest it takes on my notebook only half a second to generate the star field at maximum density, but this way will be easy to change it if its performance becomes more important.